### PR TITLE
fix: optimize app to improve pagespeed metrics

### PR DESCRIPTION
**Pull request by mardecera**
# Optimize App to improve Page Speed metrics

## Summary
Optimize or modify the js, ts, css, images, fonts code for performance metrics, accessibility, good practices, SEO to improve the [Page Speed](https://pagespeed.web.dev) ​​score.

### Problem
The app has an average score on some Page Speed ​​parameters.

### Cause
Poor optimization in fonts, images, accessibility attributes for people with virtual assistants, and metadata.

### Solution
- Split `index.js` code into heavy components using `lazy` and `suspense`.
- Use [`vite-plugin-image-optimizer`](https://www.npmjs.com/package/vite-plugin-image-optimizer) for image optimization, svg, among others (only if possible).
- Add lazy loading of images.
- Descriptive aria-labels have been added to elements that do not have descriptive text (Accessibility).
- Changed h4 element in footer to a span with the same style (Accessibility).
- Added [Helmet Async](https://www.npmjs.com/package/react-helmet-async) metadata (SEO).
- Changed loading of Google Fonts through `link` with local loading of said optimized fonts with a fallback added by the [fontaine](https://github.com/unjs/fontaine) package to avoid FOUT (flash of unstyled text).
- Optimized images for a smaller size for loading on mobile devices
- Added optional loading of images using `<picture>`, `<source>` and `media queries` for background images.
- Added a `.vscode` gitignore exclusion for my personal plugin configuration in this project.
- Added `tslib` and `typescript` packages to improve the compilation of ts code to js.
- Packages that are not needed in production have been moved to development dependencies (@testing-library/jest-dom, @vitejs/plugin-react").
- Added the corresponding `width` and `height` to images and svg that were imported as images.
- To avoid unnecessary image loading on mobile devices by the Hero component, a super small 1x1 transparent image was used with a tiny request, based on this [article](https://medium.com/@mike_masey/how-to-use-the-picture-element-to-prevent-images-loading-on-mobile-devices-1376e33b190e).

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor
- [ ] Chore

## Checklist:

These items must be completed before the code review

- [ ] Check that branch is pointing to production environment
- [x] Remove console.logs
- [x] Check imports format
- [x] Check variables destructure
- [x] Check docs
- [ ] Review the entire pull request yourself before sending it to the reviewer
- [x] Check task requirements on Monday or Sprint Planning sheet

#### Mobile
home-mobile
#### PC
home

#### Mobile
about-us

#### PC
about-us-

#### Mobile
contact-us-mobile

#### PC
contact-us

#### Mobile
faq-mobile

#### PC
faq

#### Mobile
404-mobile

### CODE REVIEW

**Review by CrhistianT7 — APPROVED** [review]
Great Job!!!

**Comment by GustavoUT22** [comment]
This branch has conflicts in the footer, please try to fix it before

File: src/pages/Clients/Clients.tsx [hunk 1]
```diff
@@ -21,6 +21,7 @@ import { IClientCard } from 'types/components/clients'
 import { IData } from 'types/components/pagination'
 import H2 from 'ui/Titles/H2'
 import useIntlMessages from 'hooks/useIntlMessages'
+import { LazyLoadImage } from 'react-lazy-load-image-component'
```
Comment (GustavoUT22) [inline src/pages/Clients/Clients.tsx:24]: Change location of LazyLoadImage import

File: src/pages/Clients/ClientCards/ClientCards.tsx [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { Link } from 'react-router-dom'
 import { CardsWrapper } from './ClientCards.styles'
 import { IClientCard } from 'types/components/clients'
 import { CardWrapper, ExternalLink } from './ClientCards.styles'
+import { LazyLoadImage } from 'react-lazy-load-image-component'
```
Comment (GustavoUT22) [inline src/pages/Clients/ClientCards/ClientCards.tsx:6]: Change location of LazyLoadImage import

File: src/pages/Blog/Subscribe/Subscribe.tsx [hunk 3]
```diff
@@ -5,6 +5,7 @@ import Section from 'ui/Section/Section'
 import Input from 'ui/input/Input'
 import { SubscribeWrapper } from './Subscribe.styles'
 import subscribeImg from './../../../assets/images/imgBlog/subscribe.svg'
+import { LazyLoadImage } from 'react-lazy-load-image-component'
```
Comment (GustavoUT22) [inline src/pages/Blog/Subscribe/Subscribe.tsx:8]: Change location of LazyLoadImage import

File: src/pages/ContactUs/ContactUs.tsx [hunk 4]
```diff
@@ -1,19 +1,42 @@
...
 import {
   ContentWrapper,
   SectionContactUs,
 } from 'components/ContactUs/ContactUs.styles'
-import ContactUsInformation from 'components/ContactUs/ContactUsInformation/ContactUsInformation'
-import ContactUsForm from 'components/ContactUs/ContactUsForm/ContactUsForm'
 import useIntlMessages from 'hooks/useIntlMessages'
 import H1 from 'ui/Titles/H1'
+import imageLinkPreview from '../../assets/images/image-link-preview.png'
+import { lazy, Suspense } from 'react'
```
Comment (GustavoUT22) [inline src/pages/ContactUs/ContactUs.tsx:11]: You need white space here between local and external imports

File: src/components/Clients/Brandings.tsx [hunk 5]
```diff
@@ -3,6 +3,7 @@ import { useEffect, useState } from 'react'
 import naturalEcoFriendly from './../../assets/images/naturaleza-ecoamigable.svg'
 import oasisRestaurant from './../../assets/images/logo-oasis-op.webp'
 import { BrandingsWrapper } from './Clients.styles'
+import { LazyLoadImage } from 'react-lazy-load-image-component'
```
Comment (GustavoUT22) [inline src/components/Clients/Brandings.tsx:6]: You need white space between local and external imports

**Review by GustavoUT22 — CHANGES_REQUESTED** [review]

File: src/pages/Clients/Clients.tsx (hunk 1 above)
Comment (mardecera) [reply src/pages/Clients/Clients.tsx:24]: Done!

File: src/pages/Clients/ClientCards/ClientCards.tsx (hunk 2 above)
Comment (mardecera) [reply src/pages/Clients/ClientCards/ClientCards.tsx:6]: Done!

File: src/pages/Blog/Subscribe/Subscribe.tsx (hunk 3 above)
Comment (mardecera) [reply src/pages/Blog/Subscribe/Subscribe.tsx:8]: Done!

File: src/components/Clients/Brandings.tsx (hunk 5 above)
Comment (mardecera) [reply src/components/Clients/Brandings.tsx:6]: Done!

File: src/pages/ContactUs/ContactUs.tsx (hunk 4 above)
Comment (mardecera) [reply src/pages/ContactUs/ContactUs.tsx:11]: Done!

**Review by GustavoUT22 — APPROVED** [review]
Nice!

**Pull request merged**